monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2619,168 @@ def _get_params(
+        fix_num_patches: number of patches to return. Defaults to None, which returns all the available patches.
+        overlap: the amount of overlap of neighboring patches in each dimension (a value between 0.0 and 1.0).
+            If only one float number is given, it will be applied to all dimensions. Defaults to 0.0.
+        sort_key: a callable or string that defines the order of the patches to be returned. If it is a callable, it
+            will be passed directly to the `key` argument of `sorted` function. The string can be "min" or "max",
+            which are, respectively, the minimum and maximum of the sum of intensities of a patch across all dimensions
+            and channels. Also "random" creates a random order of patches.
+            By default no sorting is being done and patches are returned in a row-major order.
+        pad_mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
+            ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
+            One of the listed string values or a user supplied function. Defaults to ``"wrap"``.
+            See also: https://numpy.org/doc/stable/reference/generated/numpy.pad.html
+        pad_opts: padding options, see `numpy.pad`
+        seed: random seed to generate offsets
+
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        patch_size: Sequence[int],
+        min_start_pos: Optional[Union[Sequence[int], int]] = None,
+        max_start_pos: Optional[Union[Sequence[int], int]] = None,
+        fix_num_patches: Optional[int] = None,
+        overlap: Union[Sequence[float], float] = 0.0,
+        sort_key: Optional[Union[Callable, str]] = None,
+        pad_mode: Union[NumpyPadMode, str] = NumpyPadMode.CONSTANT,
+        pad_opts: Optional[Dict] = None,
+        seed: int = 0,
+    ):
+        super().__init__(
+            patch_size=patch_size,
+            start_pos=(),
+            fix_num_patches=fix_num_patches,
+            overlap=overlap,
+            sort_key=sort_key,
+            pad_mode=pad_mode,
+            pad_opts=pad_opts,
+        )
+        self.set_random_state(seed)
+        self.min_start_pos = min_start_pos
+        self.max_start_pos = max_start_pos
+
+    def __call__(self, array: NdarrayOrTensor):
+        if self.min_start_pos is None:
+            min_start_pos = (0,) * len(self.patch_size)
+        else:
+            min_start_pos = ensure_tuple_rep(self.min_start_pos, len(self.patch_size))
+        if self.max_start_pos is None:
+            max_start_pos = tuple(s % p for s, p in zip(array.shape[1:], self.patch_size))
+        else:
+            max_start_pos = ensure_tuple_rep(self.max_start_pos, len(self.patch_size))
+
+        self.start_pos = tuple(
+            self.R.randint(low=low, high=high + 1) for low, high in zip(min_start_pos, max_start_pos)
+        )
+        return super().__call__(array)
```

Comment:
I feel maybe it's better to decouple the GridPatch and RandGridPatch transforms.
Refer to: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L540.
What do you think?

Thanks.

Reply:
May I ask what is the benefit of that? Anyways, `RandGridPatch` will depends on `GridPatch` but in this way at least we don't need to create an `GridPatch` object for each call of `RandGridPatch`, right?

Reply:
I think it's not a strict requirement so far.
@ericspod @rijobro @wyli May I know you guys' opinions?
If you guys prefer to the `inherit` way for this transform, I am OK.

Thanks.

Reply:
This inheritance pattern is the better way of doing things for now, the issue is that `RandGridPatch` is not substitutable with `GridPatch` which is perhaps a theoretical problem more than a practical one. I would rather avoid creating objects just to throw them out again (poltergeists).

In discussion with @rijobro and @atbenmurray we should really be going in the direction Pytorch does with it's `functional` submodules where the actual meat of computation is defined in functions. Classes then call these functions in their `__call__` or `forward` methods and so are often thin wrappers around the functions. In this instance the body of `GridPatch.__call__` would be put into a function with a large number of arguments in place of class members, and then it and `RandGridPatch.__call__` would call this function. The two classes wouldn't need an inheritance relationship because the reuse mechanism is moved to using functions. This is beyond the scope of this PR so I would suggest using inheritance here.

Reply:
OK, we can use `inheritance` for now, may revisit it in the future.

Thanks.